### PR TITLE
When running the same binary verify signature once

### PR DESCRIPTION
**Pull request by peregrineshahin**
SPSA tunes and same commit vs same commit runs the same binary so verify once

### CODE REVIEW

File: worker/games.py [hunk 1]
```diff
@@ -1372,16 +1372,17 @@ def parse_options(s):
...
-            new_engine,
-            run["args"]["new_signature"],
-            games_concurrency * threads,
-        )
-    except RunException as e:
-        run_errors.append(str(e))
-    except WorkerException as e:
-        raise e
+    if run["args"]["base_signature"] != run["args"]["new_signature"]:
+        try:
```
Comment (vdbergh) [inline worker/games.py:1376]: Are you sure this is ok? What if the author of the test supplies a bogus "new_signature" equal to "base_signature"? Are these signatures validated elsewhere?

Comment (peregrineshahin) [reply worker/games.py:1376]: sorry I meant `new_engine != base_engine` but I your point holds nevertheless

Comment (peregrineshahin) [reply worker/games.py:1376]: well we can check both

**Comment by peregrineshahin** [comment]
this should work
```
    if not (
        run["args"]["base_signature"] == run["args"]["new_signature"]
        and new_engine == base_engine
    ):
```

File: worker/games.py (hunk 1 above)
Comment (vdbergh) [reply worker/games.py:1379]: Yes `new_engine != base_engine` seems to be a better test!

**Comment by vdbergh** [comment]
Perhaps this PR will avoid the puzzling https://tests.stockfishchess.org/actions?max_actions=1&action=&user=&text=&before=1714917376.066326&run_id= ?

**Comment by peregrineshahin** [comment]
> Perhaps this PR will avoid the puzzling https://tests.stockfishchess.org/actions?max_actions=1&action=&user=&text=&before=1714917376.066326&run_id= ?

Yes indeed.

**Review by ppigazzini — APPROVED** [review]
Looks good w/ DEV + workers

**Comment by ppigazzini** [comment]
Triggered the workers update, thank you @peregrineshahin

**Pull request merged**